packages/react-reconciler/src/ReactFiberCommitEffects.js

```diff
@@ -146,19 +148,51 @@ export function commitHookEffectListMount(
 
           // Mount
           let destroy;
+          if (
+            enableUseResourceEffectHook &&
+            effect.kind === ResourceEffectKind
+          ) {
+            if (typeof effect.create === 'function') {
+              effect.resource = effect.create();
+              if (__DEV__) {
+                if (effect.resource == null) {
+                  console.error(
+                    'useResourceEffect must provide a callback which returns a resource. ' +
+                      'If a managed resource is not needed here, use useEffect. Received %s',
+                    effect.resource,
+                  );
+                }
+              }
+            } else if (
+              typeof effect.update === 'function' &&
+              effect.resource != null
+            ) {
+              // TODO(@poteto) what about multiple updates?
+              effect.update(effect.resource);
+            }
+            destroy = effect.destroy;
+          }
           if (__DEV__) {
             if ((flags & HookInsertion) !== NoHookEffect) {
               setIsRunningInsertionEffect(true);
             }
-            destroy = runWithFiberInDEV(finishedWork, callCreateInDEV, effect);
+            if (effect.kind === SimpleEffectKind) {
```

Comment:
Ditto here for the feature flag behavior

Reply:
this one is still needed because i unconditionally change the shape of the Effect type, so we do still need this check. i could probably console.error or something if the flag is disabled and the effect kind isn't the simple one